example/app/components/LoginScreen.tsx

```diff
@@ -3,53 +3,21 @@ import {View, Text, StyleSheet, TextInput, Pressable} from 'react-native';
 import colors from '../styles/colors';
 import {shadows} from '../styles/shadows';
 import {buttonStyles} from '../styles/button';
-import {Realm, useApp} from '@realm/react';
-
-export enum AuthState {
-  None,
-  Loading,
-  LoginError,
-  RegisterError,
-}
+import {AuthOperationName, useAuth, useEmailPasswordAuth} from '@realm/react';
 
 export const LoginScreen = () => {
-  const app = useApp();
+  const {result, logInWithEmailPassword} = useAuth();
+  const {register} = useEmailPasswordAuth();
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  const [authState, setAuthState] = useState(AuthState.None);
-
-  // If the user presses "login" from the auth screen, try to log them in
-  // with the supplied credentials
-  const handleLogin = useCallback(async () => {
-    setAuthState(AuthState.Loading);
-    const credentials = Realm.Credentials.emailPassword(email, password);
-    try {
-      await app.logIn(credentials);
-      setAuthState(AuthState.None);
-    } catch (e) {
-      console.log('Error logging in', e);
-      setAuthState(AuthState.LoginError);
-    }
-  }, [email, password, setAuthState, app]);
-
-  // If the user presses "register" from the auth screen, try to register a
-  // new account with the  supplied credentials and login as the newly created user
-  const handleRegister = useCallback(async () => {
-    setAuthState(AuthState.Loading);
-
-    try {
-      // Register the user...
-      await app.emailPasswordAuth.registerUser({email, password});
-      // ...then login with the newly created user
-      const credentials = Realm.Credentials.emailPassword(email, password);
-
-      await app.logIn(credentials);
-      setAuthState(AuthState.None);
-    } catch (e) {
-      console.log('Error registering', e);
-      setAuthState(AuthState.RegisterError);
-    }
-  }, [email, password, setAuthState, app]);
+
+  const handleLogin = useCallback(() => {
+    logInWithEmailPassword({email, password});
+  }, [logInWithEmailPassword, email, password]);
+
+  const handleRegister = useCallback(() => {
+    register({email, password});
+  }, [register, email, password]);
```

Comment:
That's a good idea.  I'll just inline the methods where they are invoked.  They are already wrapped in a useCallback internally

Reply:
I'm trying to think of a way to do this where these functions don't update.  If I make it argument based, then below I will have to do 
```
onPress={() => handleRegister({email, password})}
```
which will also reinstatiate on every state change.  If the state were implicitly passed to the `handleRegister`, this would work, but unfortunately, that's not how it works :D

Reply:
As per our discussion, since there is no possible way to avoid having this function reinstantiate, we will go the path of less characters and just inline the function to the `onPress` event.

Reply:
Agreed 🙂